test/integration/project-management.spec.ts

```diff
@@ -0,0 +1,232 @@
+      const newDisplayName = generateUniqueAppDisplayName();
+      return androidApp.setDisplayName(newDisplayName)
+          .then(() => androidApp.getMetadata())
+          .then((appMetadata) => {
+            expect(appMetadata.displayName).to.equal(newDisplayName);
+          });
+    });
+  });
+
+  describe('iosApp.setDisplayName()', () => {
+    it('successfully sets iOS app\'s display name', () => {
+      const newDisplayName = generateUniqueAppDisplayName();
+      return iosApp.setDisplayName(newDisplayName)
+          .then(() => iosApp.getMetadata())
+          .then((appMetadata) => {
+            expect(appMetadata.displayName).to.equal(newDisplayName);
+          });
+    });
+  });
+
+  describe('androidApp.{get,add,delete}ShaCertificate()', () => {
+    it('successfully gets, adds, and deletes SHA certificates', () => {
+      // Steps:
+      //   1. Check that this app has no certs.
+      //   2. Add a cert to this app.
+      //   3. Check that the cert was added successfully.
+      //   4. Delete the cert we just created.
+      //   5. Check that this app has no certs.
+      return androidApp.getShaCertificates()
+          .then((certs) => {
+            expect(certs.length).to.equal(0);
+
+            const shaCertificate = admin.projectManagement().shaCertificate(SHA_256_HASH);
+            return androidApp.addShaCertificate(shaCertificate);
+          })
+          .then(() => androidApp.getShaCertificates())
+          .then((certs) => {
+            expect(certs.length).to.equal(1);
+            expect(certs[0].shaHash).to.equal(SHA_256_HASH);
+            expect(certs[0].certType).to.equal('sha256');
+            expect(certs[0].resourceName).to.not.be.empty;
+
+            return androidApp.deleteShaCertificate(certs[0]);
+          })
+          .then(() => androidApp.getShaCertificates())
+          .then((certs) => {
+            expect(certs.length).to.equal(0);
+          });
+    });
+  });
+
+  describe('androidApp.getConfig()', () => {
+    it('successfully gets the Android app\'s config', () => {
+      return androidApp.getConfig().then((config) => {
+        expect(config).is.not.empty;
+        expect(config).includes(androidApp.appId);
+      });
+    });
+  });
+
+  describe('iosApp.getConfig()', () => {
+    it('successfully gets the iOS app\'s config', () => {
+      return iosApp.getConfig().then((config) => {
+        expect(config).is.not.empty;
+        expect(config).includes(iosApp.appId);
+      });
+    });
+  });
+});
+
+/**
+ * Ensures that an Android app owned by these integration tests exist. If not one will be created.
+ *
+ * @return {Promise<AndroidApp>} Android app owned by these integration tests.
+ */
+function ensureAndroidApp(): Promise<admin.projectManagement.AndroidApp> {
+  return admin.projectManagement().listAndroidApps()
+      .then((apps) => Promise.all(apps.map((app) => app.getMetadata())))
+      .then((metadatas) => {
+        const metadataOwnedByTest =
```

Comment:
```
if (metadatas.some((metadata) => isIntegrationTestApp(metadata.packageName))) {

}
```

Reply:
metadataOwnedByTest is reused inside the if statement, so I'd rather keep it in its current form instead of performing an array scan twice.